tests/test_qsim_device.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the QSimDevice class
+"""
+import math
+from unittest.mock import MagicMock, patch
+
+import cirq
+import qsimcirq
+import pennylane as qml
+from pennylane.wires import Wires
+import pytest
+import numpy as np
+
+from pennylane_cirq.qsim_device import QSimDevice
+
+
+@patch.multiple(QSimDevice, __abstractmethods__=set())
```

Comment:
Never seen that before!!

Reply:
What does it do :laughing:

Reply:
This is a way of allowing you to instantiate an abstract class for testing 🙂 Essentially, the `__abstractmethods__` method is temporarily patched to an empty set, so that the class looks to have no abstract methods